docs/learn/documentation/versioned/jobs/configuration-table.html

```diff
@@ -822,6 +828,40 @@
                 </tr>
 
                 <tr>
+                    <th colspan="3" class="section" id="streams"><a href="../container/streams.html">Streams</a></th>
+                </tr>
+
+                <tr>
+                    <td class="property" id="streams-system">streams.<span class="stream">stream-id</span>.samza.system</td>
+                    <td class="default"></td>
+                    <td class="description">
+                        The <span class="system">system-name</span> of the system on which this stream will be accessed.
+                        This property binds the stream to one of the systems defined with the property
+                        systems.<span class="system">system-name</span>.samza.factory. <br>
+                        If this property isn't specified, it is inherited from job.default.system.
+                    </td>
+                </tr>
+
+                <tr>
+                    <td class="property" id="streams-physical-name">streams.<span class="stream">stream-id</span>.samza.physical.name</td>
```

Comment:
@navina regarding to the code portability issue I referred to, I agree that my example is not the best one to illustrate it. Let me try another round: Samza today supports both HDFS and Kafka inputs. Now, let's say the user wants to process "PageViewEvent" in a Samza application, which may take the same "PageViewEvent" from HDFS or Kafka. W/o a unified logic id for the stream "PageViewEvent", the user's code has to deal w/ input is either "hdfs.file://my.dir/PageViewEvent" or "kafka.PageViewEvent". How is that portable between systems? The idea of introducing the logic stream id helps to:
1. make the stream ID as the top-level configure variable and allow user program directly refer to it
2. avoid fixation of physical stream name to the configure variable name
3. avoid fixation of a stream's configuration to a system's configuration variable name

Meanwhile, I don't think @jmakes is planning to remove the support for the old configuration here. I am proposing to deprecate the old configuration for streams to avoid the confusion, especially in the documentation for configuration table here.

We got a chance to move forward w/ our API in this round. I view configuration also as part of the API and I would rather to clean it now than keeping both s.t. the configuration for systems and streams are muddy. We can still support the old deprecated configuration for now, but need to mark those deprecated configurations clear in the web doc.

@jmakes the "re-write" I referred to in my comment was not referring to a config re-writer. I was merely referring to re-write the configuration table in the web doc w/ new config only and move the old ones to deprecated section.

As for the code check-in, I would suggest that we check this code in w/ support for both configurations, and open a new JIRA to discuss the deprecation of the old configuration for streams. We need to aware that we *have to* resolve the new JIRA before release to avoid the confusion as @navina pointed out.

My few cents.